packages/arcgis-rest-auth/src/UserSession.ts

```diff
@@ -451,17 +451,22 @@ export class UserSession implements IAuthenticationManager {
     options: IOAuth2Options,
     response: http.ServerResponse
   ) {
-    const { portal, clientId, duration, redirectUri }: IOAuth2Options = {
+    const { portal, clientId, duration, redirectUri, refreshTokenTTL, state }: IOAuth2Options = {
       ...{ portal: "https://arcgis.com/sharing/rest", duration: 20160 },
       ...options
     };
+    const expiration = refreshTokenTTL ? refreshTokenTTL : duration;
+
+    let url = `${portal}/oauth2/authorize?client_id=${clientId}&expiration=${expiration}&response_type=code&redirect_uri=${encodeURIComponent(
+      redirectUri
+    )}`;
```

Comment:
the tweak below would simplify things. am i losing anything in translation?

```suggestion
    const { portal, clientId, duration, redirectUri, refreshTokenTTL, state }: IOAuth2Options = {
      ...{ portal: "https://arcgis.com/sharing/rest", refreshTokenTTL: 20160 },
      ...options
    };
 
    let url = `${portal}/oauth2/authorize?client_id=${clientId}&expiration=${refreskTokenTTL}&response_type=code&redirect_uri=${encodeURIComponent(
      redirectUri
    )}`;
```

Reply:
I was thinking backwards compatibility... since the `duration` param previously set the refresh token expiration on the server, it should still be respected if the developer specified it, but update the default behavior to respect the `refreshTokenTTL` since if set this is likely the intent

Reply:
makes sense. thanks for humoring me.